airflow/models/baseoperator.py

```diff
@@ -575,6 +579,11 @@ class derived from this one results in the creation of a task object,
         significantly speeding up the task creation process as for very large
         DAGs. Options can be set as string or using the constants defined in
         the static class ``airflow.utils.WeightRule``
+    :param priority_weight_strategy: weighting method used for the effective total priority weight
```

Comment:
I do not fully understand why we need to rename the field. Even if we are using a custom implementation, is the old name not matching anymore? Especially as the same parameters apply.
In this way we force all customers to change the DAG definitions as we migrate the parameter name, in 95% of cases the parameter and function does not change.